The current wording of the error string is a bit confusing, as it might
be interpreted to mean that there's no key in the package. Can't hurt to
clarify that the public key is really just missing on the local machine:

diff --git a/python/rpm/transaction.py b/python/rpm/transaction.py
index db7ca67..7aa26b9 100644
--- a/python/rpm/transaction.py
+++ b/python/rpm/transaction.py
@@ -153,7 +153,7 @@ class TransactionSet(TransactionSetCore):
         res, msg = TransactionSetCore.hdrCheck(self, blob)
         # generate backwards compatibly broken exceptions
         if res == rpm.RPMRC_NOKEY:
-            raise rpm.error("public key not available")
+            raise rpm.error("public key not installed")
         elif res == rpm.RPMRC_NOTTRUSTED:
             raise rpm.error("public key not trusted")
         elif res != rpm.RPMRC_OK:
@@ -163,7 +163,7 @@ class TransactionSet(TransactionSetCore):
         res, h = TransactionSetCore.hdrFromFdno(self, fd)
         # generate backwards compatibly broken exceptions
         if res == rpm.RPMRC_NOKEY:
-            raise rpm.error("public key not available")
+            raise rpm.error("public key not installed")
         elif res == rpm.RPMRC_NOTTRUSTED:
             raise rpm.error("public key not trusted")
         elif res != rpm.RPMRC_OK:
